src/sysfs.rs

```diff
@@ -177,4 +189,29 @@ impl GpioIn for SysFsGpioInput {
             }
         }
     }
+
+    fn set_edge(&mut self, edge: GpioEdge) -> Result<(), Self::Error> {
+        fs::OpenOptions::new()
+            .write(true)
+            .open(format!("/sys/class/gpio/gpio{}/edge", self.gpio.gpio_num))?
+            .write_all(match edge {
+                GpioEdge::None => b"none",
+                GpioEdge::Rising => b"rising",
+                GpioEdge::Falling => b"falling",
+                GpioEdge::Both => b"both",
+            })?;
+        Ok(())
+    }
+
+    fn wait_for_edge(&mut self, timeout_ms: u64) -> Result<Option<GpioValue>, Self::Error> {
+        let _ = self.read_value()?;
+        let event = EpollEvent::new(EpollFlags::EPOLLPRI | EpollFlags::EPOLLET, 0);
+        let mut events: [EpollEvent; 1] = [event];
+        let cnt = epoll::epoll_wait(self.epoll_fd, &mut events, timeout_ms as isize)
+            .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+        Ok(match cnt {
+            0 => None, // timeout
+            _ => Some(self.read_value()?),
```

Comment:
Hmm, this whole function does three syscalls. Can we reduce the interface somewhat? E.g. if edges are set to falling, we know that the new value is low and the poller might not be interested in the result.